junifer/storage/hdf5.py

```diff
@@ -0,0 +1,954 @@
+    ) -> Dict[str, Any]:
+        """Read data (should not be called directly).
+
+        Parameters
+        ----------
+        md5 : str
+            The MD5 used as the HDF5 group name.
+        element : dict, optional
+            The element as dictionary (default None).
+
+        Returns
+        -------
+        dict
+            The retrieved data.
+
+        Raises
+        ------
+        IOError
+            If HDF5 file or data does not exist.
+
+        """
+        # Get correct URI for element;
+        # is different from uri if single_output is False
+        uri = self._fetch_correct_uri_for_io(element=element)
+
+        try:
+            logger.info(f"Loading HDF5 data for {md5} from: {uri}")
+            data = read_hdf5(
+                fname=uri,
+                title=md5,
+                slash="ignore",
+            )
+        except IOError:
+            raise_error(
+                msg=f"HDF5 file not found at: {uri}",
+                klass=IOError,
+            )
+        except ValueError:
+            raise_error(
+                msg=f"`{md5}` not found in: {uri}",
+                klass=IOError,
+            )
+        else:
+            logger.info(f"Loaded HDF5 data for {md5} from: {uri}")
+            return data
+
+    def read_df(
+        self,
+        feature_name: Optional[str] = None,
+        feature_md5: Optional[bool] = None,
+    ) -> pd.DataFrame:
+        """Read feature into a pandas.DataFrame.
+
+        Either one of ``feature_name`` or ``feature_md5`` needs to be
+        specified.
+
+        Parameters
+        ----------
+        feature_name : str, optional
+            Name of the feature to read (default None).
+        feature_md5 : str, optional
+            MD5 hash of the feature to read (default None).
+
+        Returns
+        -------
+        pandas.DataFrame
+            The features as a dataframe.
+
+        Raises
+        ------
+        IOError
+            If HDF5 file does not exist.
+
+        """
+        # Parameter conflict
+        if feature_md5 and feature_name:
+            raise_error(
+                msg=(
+                    "Only one of `feature_name` or `feature_md5` can be "
+                    "specified."
+                )
+            )
+        # Parameter absence
+        elif not feature_md5 and not feature_name:
+            raise_error(
+                msg=(
+                    "At least one of `feature_name` or `feature_md5` "
+                    "must be specified."
+                )
+            )
+        # Parameter check pass
+        else:
+            # Read metadata
+            metadata = read_hdf5(
+                fname=str(self.uri.resolve()),  # type: ignore
+                title="meta",
+                slash="ignore",
+            )
+            # Initialize MD5 variable
+            md5: str = ""
+
+            # Consider feature_md5
+            if feature_md5 and not feature_name:
```

Comment:
from the above conditions, if `feature_md5` is not none, then `feature_name` is none. Can you simplify the if conditions here?

Indeed the only condition you need to check is to get the feature_md5 from the name.

Reply:
Sorry I don't understand the reason. I am making it as explicit as possible.

Reply:
I meant that before this point, you already checked that only one of `feature_md5` or `feature_name` are not none. There's no need for the `and not feature_name` here.

Reply:
Addressed in `25643d91`.